task_sdk/src/airflow/sdk/execution_time/task_runner.py

```diff
@@ -423,13 +441,14 @@ def run(ti: RuntimeTaskInstance, log: Logger):
             state=TerminalTIState.FAIL_WITHOUT_RETRY,
             end_date=datetime.now(tz=timezone.utc),
         )
-
         # TODO: Run task failure callbacks here
-    except AirflowTaskTimeout:
-        # TODO: handle the case of up_for_retry here
-        # TODO: coagulate this exception handling with AirflowException
-        # once https://github.com/apache/airflow/issues/45307 is handled
-        ...
```

Comment:
Coagulating both these exceptions into a single handler, as they do the same thing.